sips/sip-webauthn.md

```diff
@@ -0,0 +1,160 @@
+
+![Flow A](../assets/sip-webauthn/flow-a.png)
+
+#### Flow (B) - desktop browser + security key cross-platform authentication
+
+This flow describes how a user can use a security key (e.g., Yubikey) to sign a transaction initiated on a desktop browser. The passkey is stored on the security key and is protected by a PIN or biometrics. Once the passkey is generated on the security key, it can never leave the device and it's not possible to back it up. It's advised to only use this method in a multisig setup where the security key is only one of the possible signers because if the security key is lost the account can't be recovered.
+
+The figure below describes the flow. A transaction is initiated by a web3 app which opens a pop-up window of the wallet web app (as described in the previous section) or browser extension. When the user reviews and approves the transaction (fig. B.1), the browser creates a pop-up prompting the user to authenticate using their security key (fig. B.2). In case the security key doesn't support biometrics, the user will be required to enter their PIN (fig. B.3). The user then authenticates by touching the security key (fig. B.4), and in case the security key supports biometrics, user's fingerprint is scanned. The transaction will then be signed by the passkey stored on the security key and the signature passed over to the desktop browser. The wallet app running in the browser will then broadcast the transaction to the blockchain.
+
+![Flow B](../assets/sip-webauthn/flow-b.png)
+
+#### Flow (C) - platform authentication
+
+This flow describes how a user can use a passkey to sign a transaction initiated on the same device. This flow is similar to flow (A) but without the intermediate step of scanning the QR code. It is possible on platforms that natively support passkey management (e.g., Android, iOS, macOS, Windows).
+
+The figure below describes how a user would use their phone to sign a transaction initiated in a browser on the same phone. A transaction is initiated by a mobile app or web app which opens a pop-up window of a wallet web app. When the user reviews and approves the transaction (fig. C.1), the phone creates a pop-up window requiring the user to authenticate using a pin or biometrics (fig. C.2). When the user authenticates, the transaction will be signed by the passkey and the signature passed over back to the wallet web app (C.3). The wallet app will then broadcast the transaction to the blockchain.
+
+![Flow C](../assets/sip-webauthn/flow-c.png)
+
+## Specification
+
+A new `webauthn` signature scheme is introduced that allows clients to construct transaction signatures using WebAuthn authenticators. Although the WebAuthn supports multiple signature algorithms, only the `secp256r1` algorithm is supported by Sui.
```

Comment:
In the future, Sui may support additional signature algorithms that WebAuthn also supports (besides secp256r1), so perhaps it would make sense to give the scheme a more specific name than `webauthn`?

Reply:
Hmm, it would make sense to make the scheme name specific to `secp256r1`, but by "name" what do you mean exactly? Name and description of the SIP? The data structure and function names used in the implementation? This SIP doesn't make any suggestions for the latter.